### ISSUE TITLE
useSharedValue is undefined

### ISSUE THREAD

**Issue by Adriein** [software-mansion/react-native-reanimated#1490]
## Description

After following the instructions on https://docs.expo.io/versions/latest/sdk/reanimated/ for enabling react-native-reanimated v2 on expo and trying to use useSharedValue the next error appears:

TypeError: (0, _reactNativeReanimated.useSharedValue) is not a function. (In '(0, _reactNativeReanimated.useSharedValue)(0)', '(0, _reactNativeReanimated.useSharedValue)' is undefined)   
- node_modules\react-native\Libraries\LogBox\LogBox.js:148:8 in registerError
- node_modules\react-native\Libraries\LogBox\LogBox.js:59:8 in errorImpl
- node_modules\react-native\Libraries\LogBox\LogBox.js:33:4 in console.error
- node_modules\expo\build\environment\react-native-logs.fx.js:27:4 in error
- node_modules\react-native\Libraries\Core\ExceptionsManager.js:104:6 in reportException
- node_modules\react-native\Libraries\Core\ExceptionsManager.js:171:19 in handleException
- node_modules\react-native\Libraries\Core\setUpErrorHandling.js:24:6 in handleError
- node_modules\expo-error-recovery\build\ErrorRecovery.fx.js:9:32 in ErrorUtils.setGlobalHandler$argument_0
- node_modules\regenerator-runtime\runtime.js:63:36 in tryCatch
- node_modules\regenerator-runtime\runtime.js:293:29 in invoke
- node_modules\regenerator-runtime\runtime.js:63:36 in tryCatch
- node_modules\regenerator-runtime\runtime.js:154:27 in invoke
- node_modules\regenerator-runtime\runtime.js:164:18 in PromiseImpl.resolve.then$argument_0
- node_modules\react-native\node_modules\promise\setimmediate\core.js:37:13 in tryCallOne
- node_modules\react-native\node_modules\promise\setimmediate\core.js:123:24 in setImmediate$argument_0
- node_modules\react-native\Libraries\Core\Timers\JSTimers.js:130:14 in _callTimer
- node_modules\react-native\Libraries\Core\Timers\JSTimers.js:181:14 in _callImmediatesPass
- node_modules\react-native\Libraries\Core\Timers\JSTimers.js:441:30 in callImmediates
- node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:387:6 in __callImmediates
- node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:135:6 in __guard$argument_0
- node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:364:10 in __guard
- node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:134:4 in flushedQueue
* [native code]:null in flushedQueue
* [native code]:null in invokeCallbackAndReturnFlushedQueue

### Screenshots

file where i'm using it:
![image](https://user-images.githubusercontent.com/42537962/100547917-bd129a80-3269-11eb-925e-9cc0fedbf2e8.png)

pckage.json deps:
![image](https://user-images.githubusercontent.com/42537962/100547951-e29fa400-3269-11eb-84d1-8b0207d6a8a3.png)

babel config:
![image](https://user-images.githubusercontent.com/42537962/100547955-f21eed00-3269-11eb-8a3c-2998648819db.png)

app.json:
![image](https://user-images.githubusercontent.com/42537962/100547973-02cf6300-326a-11eb-9995-93e84ee6f51b.png)

## Steps To Reproduce

1. create a react native app with expo
2. follow this guide to enable react native reanimated v2 : https://docs.expo.io/versions/latest/sdk/reanimated/
3. try to use useSharedValue on any file

### Expected behavior

I expect to use useSharedValue to update the x value while swapping

### Actual behavior

The app crashes due finding an existing function undefined

## Package versions

- React:
- React Native:
- React Native Reanimated:
- NodeJS:

**Comment by jakub-gonet** [comment]
Could you try installing Rea2 via yarn and report back here? I think I've seen similar issues in the past when using npm instead of yarn.

**Comment by vcellu** [comment]
I am having a similar issue with VSCode.   I am using Yarn.  Same project works fine outside of VSCode by just running `yarn iOS`

```
Launching the application
Attaching to the application
Starting debugger app worker.
Established a connection with the Proxy (Packager) to the React Native application
Debugger worker loaded runtime on port 36395
Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'NativeReanimated' could not be found. Verify that a module by this name is registered in the native binary.
node_modules/react-native/Libraries/Core/ExceptionsManager.js:179
Running application ModernBottomSheet ({
    initialProps =     {
    };
    rootTag = 11;
})
node_modules/react-native/Libraries/Utilities/RCTLog.js:47
Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
node_modules/react-native/Libraries/Core/ExceptionsManager.js:179
Running surface LogBox ({
    initialProps =     {
    };
    rootTag = 21;
})
node_modules/react-native/Libraries/Utilities/RCTLog.js:47
Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
node_modules/react-native/Libraries/Core/ExceptionsManager.js:179
Failed to mount LogBox within 1s

```

**Comment by Adriein** [comment]
> Could you try installing Rea2 via yarn and report back here? I think I've seen similar issues in the past when using npm instead of yarn.

I'm gonna try it tomorrow but why is this happening? I mean, npm is also a package manager, I've never had problems installing packages via npm so why is this library different? Btw thanks a lot for the quick response jakub 💯

**Comment by Adriein** [comment]
I've just tried it with yarn, same error... I've seen another opened issues having the same problem

**Comment by AtheerAPeter** [comment]
same issue here

**Comment by karo-dc** [comment]
Also I have the same problem, can not use v2 package, useSharedValue is undefined. Maybe someone solved this?

**Comment by Aaron-Porter** [comment]
Running into this same problem. Configured according to the expo guide. Receiving this error:

`(0, _reactNativeReanimated.useSharedValue) is not a function. (In '(0, _reactNativeReanimated.useSharedValue)(0)', '(0, _reactNativeReanimated.useSharedValue)' is undefined)`

**Comment by ajlamarc** [comment]
https://docs.expo.io/versions/latest/sdk/reanimated/
I had this problem and it just resolved luckily, here was my steps:

1.  From the link above, update the babel config file as shown.
2.  expo install react-native-reanimated (downgraded from 2.0.0 to 1.1.3)
3.  yarn add react-native-reanimated@2.0.0-rc.0
4.  expo start -c

Perhaps removing the package and starting over was enough to reset it and wire it up properly.

**Comment by JosePatricio** [comment]
still facing this error using react native cli , anyone solved it?

**Comment by sinclaireric** [comment]
the error persist

**Comment by guymross** [comment]
Any progress on this?

**Comment by svpease** [comment]
> Any progress on this?

After learning much more about tsconfig.js than I probably needed to, I discovered that I could finally fix my problem simply by running `npx react-native --reset-cache;`

In addition, make sure that you are updating the babel.config.js file as it describes in their [installation documentation](https://docs.swmansion.com/react-native-reanimated/docs/installation/). I also overlooked where it says running `yarn add react-native-reanimated@next;` and at the moment my brain is too mushy at the moment to evaluate whether that's what really fixed it.

In conclusion,

1. update the babel.config.js file described via their [installation documentation](https://docs.swmansion.com/react-native-reanimated/docs/installation/)
2. make sure you clear the cache: `npx react-native --reset-cache;`
2. install react-native-reanimated@next as a dependency, which can be done via: `npm i react-native-reanimated@next;`
3. and if using iOS, don't forget to install pods: `cd ios && pod install && cd ..;`

Hope this helps you.

**Comment by piaskowyk** [comment]
Hey @Adriein 👋 Does the problem still occurs after the update to the newest version Reanimated and after clearing the cache? If yes, could you prepare an example app to reproduce your issue?

**Comment by j-piasecki** [comment]
Hello @Adriein. Could you check if updating to the newest version of Reanimated and/or clearing cache solves the problem? It it doesn't, may you prepare a repository with a reproduction?

**Comment by ZeroPie** [comment]
Can confirm this is stil broken on "react-native-reanimated": "^2.8.0",
